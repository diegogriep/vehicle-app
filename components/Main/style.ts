'use client'

import styled from 'styled-components'

export const Wrapper = styled.main`
  background-color: #f0f0f0;
  color: white;
  font-size: 20px;
`
