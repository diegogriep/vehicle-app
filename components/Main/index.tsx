import * as S from './style'

const Main = () => (
  <S.Wrapper>
    <h1>Vehicles</h1>
  </S.Wrapper>
)

export default Main
