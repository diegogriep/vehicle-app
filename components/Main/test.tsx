import { render, screen } from '@testing-library/react'
import Main from '.'

describe('<Main />', () => {
  it('should render the component', () => {
    const { container } = render(<Main />)

    expect(
      screen.getByRole('heading', { name: /vehicle/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the colors correctly', () => {
    const { container } = render(<Main />)

    expect(container.firstChild).toHaveStyle({ 'background-color': '#f0f0f0' })
  })
})
