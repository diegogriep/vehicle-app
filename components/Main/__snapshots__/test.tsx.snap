// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the component 1`] = `
.c0 {
  background-color: #f0f0f0;
  color: white;
  font-size: 20px;
}

<main
  class="c0"
>
  <h1>
    Vehicles
  </h1>
</main>
`;
